File: python-sdk/cleanster/models/payment_method.py
"""PaymentMethod model."""

from typing import Any, Dict, Optional


class PaymentMethod:
    """Represents a saved payment method (card or PayPal)."""

    def __init__(self, data: Dict[str, Any]):
        self.id: Optional[int] = data.get("id")
        self.type: Optional[str] = data.get("type")
        self.last_four: Optional[str] = data.get("lastFour")
        self.brand: Optional[str] = data.get("brand")
        self.is_default: Optional[bool] = data.get("isDefault")
        self._raw = data

    def __repr__(self) -> str:
        return (
            f"PaymentMethod(id={self.id}, type={self.type!r}, "
            f"last_four={self.last_four!r})"
        )

File: csharp-sdk/src/Cleanster/Models/PaymentMethod.cs
using System.Text.Json.Serialization;

namespace Cleanster.Models;

/// <summary>Represents a saved Stripe card or PayPal payment method.</summary>
public sealed record PaymentMethod
{
    [JsonPropertyName("id")]        public int     Id        { get; init; }
    [JsonPropertyName("type")]      public string  Type      { get; init; } = "";
    /// <summary>Last 4 digits — present for card payment methods, <see langword="null"/> for others.</summary>
    [JsonPropertyName("lastFour")]  public string? LastFour  { get; init; }
    /// <summary>Card brand (e.g., "visa", "mastercard") — <see langword="null"/> for non-card methods.</summary>
    [JsonPropertyName("brand")]     public string? Brand     { get; init; }
    [JsonPropertyName("isDefault")] public bool    IsDefault { get; init; }
}
